flink-runtime/src/main/java/org/apache/flink/runtime/util/config/memory/ProcessMemorySpecBase.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.util.config.memory;
+
+import org.apache.flink.configuration.MemorySize;
+
+/**
+ * Process memory components.
```

Comment:
I would suggest:
`Common memory components of Flink processes (e.g. JM or TM).`
also, maybe, `CommonProcessMemorySpec` is better name because it is self-sufficient on its and not really always abstract (e.g. returned from `ProcessMemoryUtils`). We just choose for now to extend it for JM/TM. I can see it also being a field of some specs, maybe in future making it a composition.